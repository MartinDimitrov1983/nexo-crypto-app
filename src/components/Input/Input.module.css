.wrapper {
    position: relative;
}

.label {
    cursor: pointer;
    display: block;
    text-transform: uppercase;
    position: absolute;
    background: #f5f6f8;
    top: 3px;
    left: 15px;
    font-size: 12px;
    padding: 0 8px;
    border-radius: 6px;
}

.input {
    height: 40px;
    outline: none;
    margin: 10px 0;
    padding: 0 10px;
    cursor: pointer;
    border: 3px solid #1e4dd8;
    border-radius: 10px;
    transition: all 0.3s ease-in-out;
    background: #f5f6f8;
}

.wrapper:hover {
    transform: scale(1.1);
}
