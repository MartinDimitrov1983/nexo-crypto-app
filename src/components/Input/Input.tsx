import React from 'react';
import styles from './Input.module.css';

interface InputProps {
    id: string;
    labelName: string;
    value: string;
    handleOnChange: Function;
}

const Input = ({ labelName, handleOnChange, value, id }: InputProps) => {
    return (
        <div className={styles.wrapper}>
            <label className={styles.label} htmlFor={id}>
                {labelName}
            </label>
            <input
                className={styles.input}
                id={id}
                type="text"
                value={value}
                onChange={(e) => handleOnChange(e)}
            />
        </div>
    );
};

export default Input;
