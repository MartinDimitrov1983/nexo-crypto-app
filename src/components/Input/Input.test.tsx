import React from 'react';
import { render } from '@testing-library/react';

import Input from './Input';

test('Snapshot Input', () => {
    const { asFragment } = render(
        <Input
            id="test"
            labelName="test"
            value="test"
            handleOnChange={() => {}}
        />,
    );
    expect(asFragment()).toMatchSnapshot();
});
