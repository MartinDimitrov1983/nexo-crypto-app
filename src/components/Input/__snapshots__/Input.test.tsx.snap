// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Snapshot Input 1`] = `
<DocumentFragment>
  <div
    class="wrapper"
  >
    <label
      class="label"
      for="test"
    >
      test
    </label>
    <input
      class="input"
      id="test"
      type="text"
      value="test"
    />
  </div>
</DocumentFragment>
`;
